src/sfptpd_clock.c

```diff
@@ -301,7 +320,7 @@ static struct sfptpd_interpolation clock_format_specifiers[] = {
 
 static inline void clock_lock(void)
 {
-	int rc = pthread_mutex_lock(sfptpd_clock_lock);
+	int rc = sfptpd_clock_lock == NULL ? 0 : pthread_mutex_lock(sfptpd_clock_lock);
```

Comment:
This seems like it would open us to bugs if we have any cases where we try to lock prior to the lock being created; since we're just saying "Ok, locked".  Which I assume is intended for cases where there's never going to be a lock to take?  Which would previously noisily fail.  Would it be better to have a separate "lock_if_available" to make that choice more explicit?

Reply:
I'm not sure there would be any benefit to that. The change here is to allow the clock module to be used in applications that are single-threaded and therefore have no need of the lock. It would be programmer error to use the clock module API in any way without providing a hardware state lock if the application were multithreaded, which it must do at module initialisation time, so the problem would then be reduced to calling functions without initialising first - a more general problem that two differently-named versions of the same functions, one without this check, wouldn't help with.